lib/psulib_traject/processors/record_type.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module PsulibTraject::Processors
+  class RecordType
+    def self.call(record:, current_formats: [])
+      if current_formats.any? && !current_formats.include?(Traject::TranslationMap.new('formats_949t')['MICROFORM'])
+        return current_formats
+      end
+
+      new_formats = new(record, current_formats).resolve
+
+      current_formats
+        .push(new_formats)
+        .compact
+    end
+
+    attr_reader :record, :current_formats
+
+    def initialize(record, current_formats)
+      @record = record
+      @current_formats = current_formats
+      freeze
+    end
+
+    def resolve
+      case record.leader[6..7]
+      when /^a/
+        bibliographic_level
+      when /^g/
+        type_of_visual_material
+      when /^t(a|m)/
+        'Archives/Manuscripts'
+      else
+        Traject::TranslationMap
+          .new('formats_leader6')
+          .translate_array([record.leader[6]])
+          .first
+      end
+    end
+
+    def bibliographic_level
```

Comment:
@awead Wondering if this could be private as well?

Reply:
I thought about this too. Technically, everything other than `:resolve` should be private. Although the public/private distinction here gets a little weird because we're using a (the?) command pattern here, and the entry point for the whole thing is `RecordType.call`. Does that mean someone would:

``` ruby
type_processor = RecordType.new(record, current_formats)
type_processor.bibliographic_level
```

🤷  Probably not... And at the very least, it's not something we need to support right now. If we make them private, that will result in the tests doing `type_processor.send(:bibliographic_level)` which is considered "bad," but I'd rather have good test coverage.

Anyway, it's an interesting conundrum, and also the result of my chosen `.call` pattern here for the processors, which we could revisit...